engine/applicationwindow.cpp

```diff
@@ -120,3 +114,55 @@ GLFWwindow* ApplicationWindow::GetWindow() const
 }
 
 } // namespace pong
+
+namespace pong::application
+{
+
+ApplicationWindow* gApplicationWindow { nullptr };
+
+ApplicationWindow* GetWindowInstance()
+{
+    return gApplicationWindow;
+}
+
+void SetWindowInstance(ApplicationWindow* window)
+{
+    gApplicationWindow = window;
+}
+
+void WindowResizeCallback(GLFWwindow* window, int width, int height)
+{
+    GetWindowInstance()->WindowResizeCallback(window, width, height);
+}
+
+int GetScreenWidth()
+{
+    return GetWindowInstance()->GetScreenWidth();
+}
+
+int GetScreenHeight()
+{
+    return GetWindowInstance()->GetScreenHeight();
+}
+
+void SetWindowTitle(const std::string& title)
+{
+    GetWindowInstance()->SetWindowTitle(title);
+}
+
+void SetWindowIcon(const std::string& iconPath)
+{
+    GetWindowInstance()->SetWindowIcon(iconPath);
```

Comment:
One of the things I liked about the Unreal version of this was it made it clear it was operating on some global statics. Can you incorporate that here somehow? Via an object or a namespace or similar?

```
GlobalStatics::WindowInstance()->
```

Makes it super painfully obvious at the call site you're mucking with globals.

Reply:
Hmm yeah what about something like
`globals::audio::PlaySound()` and `globals::application::SetWindowIcon()` under a `globals` namespace? I think that makes sense.

Reply:
yeah, just something indicating there is something going on in the globals when it happens. I think that when it's necessary to mutate global state it should be hyper obvious